### DISCUSSION TITLE
CSP package on npmjs.com

### DISCUSSION THREAD

**Discussion by chadanuk** [alpinejs/alpine#1867] [4. General]
Firstly, we're really enjoying Alpine!

However, we've run into a bit of a blocker because we want to use the CSP safe version (ideally from `node_modules`) but it doesn't appear to be in the npm package and there doesn't appear to be a separate package on npmjs.com for it.

Are there plans for the CSP safe module to be released as a package on npmjs.com?

**Comment by alexandrubau** [comment]
I am trying to develop a Chrome Extension which doesn't allow `eval()` to be used. I came across the same problem, I can't find the CSP version even though it is specified in the [docs](https://alpinejs.dev/advanced/csp).

**Comment by browner12** [comment]
The CSP build has not been deployed to NPM, so the instructions are not entirely accurate.

The only way to currently get the CSP build is clone the repo, run the build, and copy/paste the resulting ESM or CJS file.

Fair warning, I've found the CSP build to be basically unusable. Even something as simple as an x-show does not appear to be working.

**Comment by sawilde** [comment]
Any chance you could publish the CSP package as an alpha; building your own copy from source can be potentially problematic for people trying to switch to the CSP build (I know I had issues getting it in to the pipeline - I gave up trying to use the cdn flavour).

I ran into a couple of gotchas but that was due to my ignorance and was eventually overcome.

**Reply by ekwoka** [reply]
It's mostly not quite ready for prime time, mainly because nobody is attempting to maintain it.

Building from source shouldn't be much of an issue. Just clone, install, build, then copy the file out.
